### ISSUE TITLE
local inter-process communication doesn't need to go via the NC channel

### ISSUE THREAD

**Issue by qnikst** [haskell-distributed/distributed-process#220]

**Comment by qnikst** [comment]
On the other hand taking NC away from local communication removes a number of possible errors with NC lock or bottoms in values that are sent.

**Comment by facundominguez** [comment]
I don't follow that argument. Maybe an example is in order.

**Comment by qnikst** [comment]
Mine? If NC is not used in local communication then unsafe send of  message that will evaluate to error will not break NC.

**Comment by mboes** [comment]
@qnikst That's not a good argument I think - more behaviour special cases is a bad thing, not a good thing.

**Comment by qnikst** [comment]
Agree.

**Comment by hyperthunk** [comment]
Yes, I agree too. The purpose of skipping the NC was to reduce latency, and avoid bottlenecks in the NC thread. I think we can probably leave this as an open question for now, as nobody is *yet* screaming at us to speed up intra-node comms.
